### PR TITLE
Update TGS DMAPI. Add updater workflow

### PR DESCRIPTION
**Pull request by Cyberboss**
For OpenDream support (TGS supports it out of the box)

Example of what the auto-update PR does https://github.com/tgstation/tgstation/pull/81655

### CODE REVIEW

**Review by ZeWaka — COMMENTED** [review]
Don't need auto updating nor did you re-fix your stuff

File: code/modules/tgs/v3210/api.dm [hunk 1]
```diff
@@ -0,0 +1,244 @@
+#define REBOOT_MODE_NORMAL 0
```
Comment (ZeWaka) [inline code/modules/tgs/v3210/api.dm:1]: I don't give a shit about ancient tgs either

Comment (Cyberboss) [reply code/modules/tgs/v3210/api.dm:1]: It's part of the package, will eventually be removed when MSO stops being a nerd about it.

**Comment by Cyberboss** [comment]
Need to release a patch before I can fix the lint errors

**Comment by Cyberboss** [comment]
https://github.com/tgstation/tgstation-server/pull/1805

File: code/modules/tgs/v3210/api.dm (hunk 1 above)
Comment (ZeWaka) [reply code/modules/tgs/v3210/api.dm:1]: I mean it's either you remove it or I commit to this branch to do so

Comment (Cyberboss) [reply code/modules/tgs/v3210/api.dm:1]: Does it really matter? These files are `#ifdef` excluded already (https://github.com/goonstation/goonstation/pull/18297/files#diff-fdad9a583d15fb729dd330a8f5e44935fd9e28c4570d0628fdad1e99635edd01R6). Modifying the code just creates an unnecessary diff between the official release.

Comment (Cyberboss) [reply code/modules/tgs/v3210/api.dm:1]: Guess this PR is sitting around till @MrStonedOne updates off V3 cause can't strip it from source until then.

**Comment by ZeWaka** [comment]
L+get evergreened+L again+EoL+cringe

**Pull request merged**